modules/apps/product-navigation/product-navigation-personal-menu-api/src/main/java/com/liferay/product/navigation/personal/menu/PersonalMenuEntry.java

```diff
@@ -59,6 +60,12 @@ public default String getIcon(PortletRequest portletRequest) {
 	 */
 	public String getLabel(Locale locale);
 
+	public default JSONObject getPortletJSOnClickConfig(
+		HttpServletRequest httpServletRequest) {
+
+		return null;
+	}
```

Comment:
So for clarity, I'd say we do one of two things:

1. Add a javadoc on this method that details which properties need to be on the object and what they're for.
2. Make this object a dedicated class (could be an inner class in this interface) and add the docs on each getter method.

Thoughts?

Reply:
> Make this object a dedicated class (could be an inner class in this interface) and add the docs on each getter method.

I'm not totally sure what you mean by this, do you have any examples you could share of what you mean here? The words make sense, but my lack of java does not help me here

Reply:
Thinking about this more...

It would be up to whatever class implements PersonalMenuEntry to define what is in this config object, so I don't know if it makes sense to have this be a class and define what type of object it should return.

Although maybe this is an anti-pattern in Java that I am unaware of.

Reply:
Is the config going to be read anywhere else except for `PersonalMenu.es.js`? If it's [just read here](https://github.com/liferay-frontend/liferay-portal/pull/2240/files/1a6318e696fe0445d9c0356b88aeddeee630dae0#diff-efbd3900646f435e4104e9897e22e1b9550726747d937322deaf934b46e8b4a1R49-R54), then I guess my goal is to avoid a dev having to trace the source to figure out which keys to put into the JSON object. Sorry if I'm misunderstanding something here.

Reply:
I'm not too picky about _how_ we accomplish that. It doesn't have to be an inner class, but it's one way to leverage the type system to "document" what's needed.